src/sparseml/exporters/transforms/gemm_to_qlinearmatmul.py

```diff
@@ -0,0 +1,252 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from onnx import ModelProto, helper, numpy_helper
+
+from sparseml.exporters.transforms.onnx_transform import OnnxTransform
+from sparseml.exporters.transforms.utils.helpers import (
+    delete_quant_node,
+    get_quantization_params,
+    quantize_array,
+)
+from sparseml.exporters.transforms.utils.matching import (
+    INITIALIZER_MATCH,
+    MatchResult,
+    get_structural_matches,
+    optional_node,
+)
+from sparseml.onnx.utils.graph_editor import (
+    ONNXGraph,
+    remove_node_and_params_from_graph,
+)
+from sparseml.onnx.utils.helpers import get_node_attributes, get_node_output_nodes
+
+
+__all__ = ["GemmToQLinearMatMul"]
+
+
+class GemmToQLinearMatMul(OnnxTransform):
+    """
+    # Variant 1
+
+    ```
+                        weight (initializer)
+                        |
+    input               QuantizeLinear
+    |                   |
+    DequantizeLinear    DequantizeLinear
+    |                   |
+            Gemm ( with no attributes)
+            |
+            QuantizeLinear
+            |
+            NOT (DequantizeLinear -> Gemm)
```

Comment:
what does NOT mean here?

Reply:
Yeah can probably add more documentation on that. Basically if there is a Q->DQ->Gemm after the first gemm, this transform does not apply. Unsure how to capture that.

i.e. you can have:
1. Gemm -> Q -> W/e
2.  Gemm -> Q -> DQ -> Not GEMM

Reply:
makes sense